agent/agent_test.go

```diff
@@ -684,9 +692,8 @@ func TestAgent(t *testing.T) {
 			Logger:     slogtest.Make(t, nil).Leveled(slog.LevelInfo),
 			Filesystem: filesystem,
 		})
-		t.Cleanup(func() {
-			_ = closer.Close()
-		})
+		defer closer.Close()
```

Comment:
For learning purposes - which benefits does this bring here (`t.Cleanup` vs `defer`)?

Reply:
Defer is more natural and reads more like Go code, so it's well suited for the actual test function/body. Using `t.Cleanup`, especially if mixed with `defer`, has a weird execution flow where all defers will run first and cleanups later. So IMO `t.Cleanup` is best reserved for helpers that set up things that need to be torn down later (without caller having to `.Close()`).

For instance, if I had only added:

```go
coordinator := tailnet.NewCoordinator()
defer coordinator.Close()
```

Then the execution flow in this function would've started by closing the `coordinator` and only later the `closer`, which is the wrong way to tear it down.